demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,503 @@
+// Copyright (C) 2018-2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <vector>
+#include <queue>
+#include <memory>
+#include <string>
+#include <chrono>
+#include <condition_variable>
+#include <mutex>
+#include <cstdio>
+#include <functional>
+#include <atomic>
+
+#include <inference_engine.hpp>
+
+#include <ie_iextension.h>
+
+#include <samples/common.hpp>
+#include <samples/slog.hpp>
+#include <samples/args_helper.hpp>
+#include <samples/ocv_common.hpp>
+
+#include <vpu/vpu_plugin_config.hpp>
+#include <cldnn/cldnn_config.hpp>
+
+#include "imagenet_classification_demo.hpp"
+#include "grid_mat.hpp"
+
+using namespace InferenceEngine;
+
+struct InferRequestInfo {
+    struct InferRequestImage {
+        cv::Mat mat;
+        unsigned rightClass;
+        std::chrono::time_point<std::chrono::steady_clock> startTime;
+    };
+
+    InferRequest &inferRequest;
+    std::vector<InferRequestImage> images;
+};
+
+bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+    // ---------------------------------Parsing and validation of input args----------------------------------
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        showUsage();
+        showAvailableDevices();
+        return false;
+    }
+
+    slog::info << "Parsing input parameters" << slog::endl;
+    if (FLAGS_i.empty()) {
+        throw std::logic_error("Parameter -i is not set");
+    }
+    if (FLAGS_m.empty()) {
+        throw std::logic_error("Parameter -m is not set");
+    }
+    if (FLAGS_labels.empty()) {
+        throw std::logic_error("Parameter -labels is not set");
+    }
+    if (FLAGS_nt < 1) {
+        throw std::runtime_error("Parameter -nt must be >= 1");
+    }
+
+    return true;
+}
+
+cv::Mat resizeImage(cv::Mat& image, int modelInputResolution) {
+    double scale = static_cast<double>(modelInputResolution) / std::min(image.cols, image.rows);
+
+    cv::Mat resizedImage;
+    cv::resize(image, resizedImage, cv::Size(), scale, scale);
+
+    cv::Rect imgROI;
+    if (resizedImage.cols >= resizedImage.rows) {
+        int fromWidth = resizedImage.cols/2 - modelInputResolution/2;
+        imgROI = cv::Rect(fromWidth,
+                          0,
+                          std::min(modelInputResolution, resizedImage.cols - fromWidth),
```

Comment:
Why `min`? The image must be a square, so width must match height.

Reply:
Both width and height should be either `resizedImage.cols - fromWidth` or `modelInputResolution`. I think `modelInputResolution` is clearer.